File: modules/executors/outputs/watch_gui.js
(function(global) {

        var LiteGraph = global.LiteGraph;
        /* ********************** *******************
                WatchNodeGui: Watch
         ********************** *******************  */
        //node constructor class
/*
        function WatchNodeGui()
        {
            this.addInput("in","string");
            this.container = new DivContainer(this)
            this.addCustomWidget( this.container);
            this.container.addWidget('text_output', '', {'property': 'parameters'})
            this.history = {};


        }

        WatchNodeGui.prototype.connectPublishers = function(subscribe) {
            // output
            var source_location = {position: "input", slot: 0}
            var eventId = {type: "output", slot: 0};
            var action = {type: "container_action", target: "set", parameter: "parameters"}
            subscribe(this, source_location, eventId, action)

            // print
            var source_location = {position: "input", slot: 0}
            var eventId = {type: "print"};
            var action = {type: "container_action", target: "append", parameter: "parameters", silent: true}
            subscribe(this, source_location, eventId, action)

            // starting -> reset
            var source_location = {position: "input", slot: 0}
            var eventId = {type: "starting"};
            var action = {type: "container_action", target: "reset", parameter: "parameters"}
            subscribe(this, source_location, eventId, action)

        }


        WatchNodeGui.title = "Watch"
        //LiteGraph.registerNodeType("outpute/watch", WatchNodeGui );
*/

        })(this);

File: modules/executors/outputs/watch_gui.py
from modules.executors.common import BaseGuiNode


class WatchNodeGui(BaseGuiNode):
    node_title = "Watch"

    def __init__(self):
        pass

    def buildNode(self):
        builder = self._initBuilder()

        builder.addInput("in", "string");

        builder.addCustomWidget("text_output", "", {"property": "parameters"})
        builder.setPath("output/watch")

        source_location = {"position": "input", "slot": 0}
        eventId = {"type": "output"};
        action = {"type": "widget_action", "target": "set", "property": "parameters", "silent": True}
        builder.subscribe(source_location, eventId, action)

        #source_location = {"position": "input", "slot": 0}
        #eventId = {"type": "print"};
        #action = {"type": "widget_action", "target": "append", "property": "parameters"}
        #builder._subscribe(source_location, eventId, action)
        builder.setCallback("onExecute", """function(){var val =this.getInputData(0); if(val) this.container.setValue("parameters",""+val)}""")

        source_location = {"position": "input", "slot": 0}
        eventId = {"type": "starting"};
        action = {"type": "widget_action", "target": "reset", "property": "parameters"}
        builder.subscribe(source_location, eventId, action)
        return builder